packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -219,6 +310,134 @@ - (UIImage *)scaleImage:(UIImage *)image by:(id)scaleParam {
+  if ([FLTGoogleMapMarkerController isScalableWithScaleFactorFromSize:originalPixelSize
+                                                               toSize:size]) {
+    // Scaled image has close to same aspect ratio,
+    // updating image scale instead of resizing image.
+    CGFloat factor = originalPixelWidth / size.width;
+    return [FLTGoogleMapMarkerController scaledImage:image withScale:(image.scale * factor)];
+  } else {
+    // Aspect ratios differ significantly, resize the image.
+    UIGraphicsImageRendererFormat *format = [UIGraphicsImageRendererFormat defaultFormat];
+    format.scale = 1.0;
+    format.opaque = NO;
+    UIGraphicsImageRenderer *renderer = [[UIGraphicsImageRenderer alloc] initWithSize:size
+                                                                               format:format];
+    UIImage *newImage =
+        [renderer imageWithActions:^(UIGraphicsImageRendererContext *_Nonnull context) {
+          [image drawInRect:CGRectMake(0, 0, size.width, size.height)];
+        }];
+
+    // Return image with proper scaling.
+    return [FLTGoogleMapMarkerController scaledImage:newImage withScale:image.scale];
+  }
+}
+
+/// Scales an input UIImage to a specified width and height preserving aspect ratio if both
+/// widht and height are not given..
+///
+/// @param image The UIImage to scale.
+/// @param width The target width to scale the image to.
+/// @param height The target height to scale the image to.
+/// @param screenScale The current screen scale.
+/// @return UIImage Returns the scaled UIImage.
++ (UIImage *)scaledImage:(UIImage *)image
+               withWidth:(NSNumber *)width
+                  height:(NSNumber *)height
+             screenScale:(CGFloat)screenScale {
+  if (!width && !height) {
+    return image;
+  }
+
+  CGFloat targetWidth = width ? width.doubleValue : image.size.width;
+  CGFloat targetHeight = height ? height.doubleValue : image.size.height;
+
+  if (width && !height) {
+    // Calculate height based on aspect ratio if only width is provided.
+    double aspectRatio = image.size.height / image.size.width;
+    targetHeight = round(targetWidth * aspectRatio);
+  } else if (!width && height) {
+    // Calculate width based on aspect ratio if only height is provided.
+    double aspectRatio = image.size.width / image.size.height;
+    targetWidth = round(targetHeight * aspectRatio);
+  }
+
+  CGSize targetSize =
+      CGSizeMake(round(targetWidth * screenScale), round(targetHeight * screenScale));
+  return [FLTGoogleMapMarkerController scaledImage:image withSize:targetSize];
+}
+
+/// Checks if an image can be scaled from an original size to a target size using a scale factor
+/// while maintaining the aspect ratio.
+///
+/// @param originalSize The original size of the image.
+/// @param targetSize The desired target size to scale the image to.
+/// @return A BOOL indicating whether the image can be scaled to the target size with scale
+/// factor.
```

Comment:
Same here; declaration comments shouldn't be duplicated at the definition when there's a separate declaration.